core/src/main/java/org/infinispan/health/HealthStatus.java

```diff
@@ -0,0 +1,24 @@
+package org.infinispan.health;
+
+/**
+ * General Health status.
+ *
+ * @author Sebastian Łaskawiec
+ * @since 9.0
+ */
+public enum HealthStatus {
+    /**
+     * Given entity is unhealthy.
+     */
+    RED,
+
+    /**
+     * Given entity is healthy.
+     */
+    GREEN,
+
+    /**
+     * Given entity is healthy but a rebalance is in progress.
+     */
+    YELLOW
```

Comment:
I would prefer to talk about such things during design phase not the implementation. 

However if others have similar thoughts, I will change it.
@galderz @gustavonalle @tristantarrant (mentioned only folks involved in this review) - thoughts?